sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/AuthToken.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Microsoft.Rest;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.ContainerRegistry
+{
+    /// <summary>
+    /// Simple authentication class for use with Token related api calls for determining scopes and other such things
+    /// </summary>
+    public class TokenCredentials : ServiceClientCredentials
+    {
+        private string _authHeader { get; set; }
+
+        /*To be used for General Login Scheme*/
+        public TokenCredentials(string username, string password)
+        {
+            _authHeader = EncodeTo64(username + ":" + password);
+        }
+
+        /*To be used for exchanging AAD Tokens for ACR Tokens*/
+        public TokenCredentials()
+        {
+            _authHeader = null;
+        }
+
+        public override async Task ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException("request");
+            }
+            if (_authHeader != null)
+            {
+                request.Headers.Authorization = new AuthenticationHeaderValue("Basic", _authHeader);
+            }
+            await base.ProcessHttpRequestAsync(request, cancellationToken);
+        }
+
+        static public string EncodeTo64(string toEncode)
```

Comment:
This code is duplicated (see above). Can this be moved to an internal helper class instead?